### PR TITLE
[MM-16562] Migrate User.GetUnreadCountForChannel to Sync by default

### PR DESCRIPTION
**Pull request by AGMETEOR**
Summary
This PR migrates User.GetUnreadCountForChannel to Sync by default

Ticket Link
[Jira Ticket](https://mattermost.atlassian.net/browse/MM-16562)

Fixes [#11394](https://github.com/mattermost/mattermost-server/issues/11394)

### CODE REVIEW

File: store/sqlstore/user_store.go [hunk 1]
```diff
@@ -1145,14 +1145,12 @@ func (us SqlUserStore) GetUnreadCount(userId string) (int64, error) {
...
-			result.Err = model.NewAppError("SqlUserStore.GetMentionCountForChannel", "store.sql_user.get_unread_count_for_channel.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = count
-		}
-	})
+func (us SqlUserStore) GetUnreadCountForChannel(userId string, channelId string) (int64, *model.AppError) {
+	if count, err := us.GetReplica().SelectInt("SELECT SUM(CASE WHEN c.Type = 'D' THEN (c.TotalMsgCount - cm.MsgCount) ELSE cm.MentionCount END) FROM Channels c INNER JOIN ChannelMembers cm ON c.Id = cm.ChannelId AND cm.ChannelId = :ChannelId AND cm.UserId = :UserId", map[string]interface{}{"ChannelId": channelId, "UserId": userId}); err != nil {
+		return count, model.NewAppError("SqlUserStore.GetMentionCountForChannel", "store.sql_user.get_unread_count_for_channel.app_error", nil, err.Error(), http.StatusInternalServerError)
+	} else {
```
Comment (jespino) [inline store/sqlstore/user_store.go]: You can remove this else because the if always returns

Comment (AGMETEOR) [reply store/sqlstore/user_store.go]: @jespino thank you. I just made the changes.

File: store/sqlstore/user_store.go [hunk 2]
```diff
@@ -1145,14 +1145,12 @@ func (us SqlUserStore) GetUnreadCount(userId string) (int64, error) {
...
-			result.Err = model.NewAppError("SqlUserStore.GetMentionCountForChannel", "store.sql_user.get_unread_count_for_channel.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = count
-		}
-	})
+func (us SqlUserStore) GetUnreadCountForChannel(userId string, channelId string) (int64, *model.AppError) {
+	count, err := us.GetReplica().SelectInt("SELECT SUM(CASE WHEN c.Type = 'D' THEN (c.TotalMsgCount - cm.MsgCount) ELSE cm.MentionCount END) FROM Channels c INNER JOIN ChannelMembers cm ON c.Id = cm.ChannelId AND cm.ChannelId = :ChannelId AND cm.UserId = :UserId", map[string]interface{}{"ChannelId": channelId, "UserId": userId})
+	if err != nil {
+		return count, model.NewAppError("SqlUserStore.GetMentionCountForChannel", "store.sql_user.get_unread_count_for_channel.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (marianunez) [inline store/sqlstore/user_store.go]: In the case of error, this should return 0 for the first parameter to be consistent.

Comment (AGMETEOR) [reply store/sqlstore/user_store.go]: @marianunez thank you for the review. I have made the changes.

**Comment by jespino** [comment]
@AGMETEOR gofmt is failing (see the jenkins error), probably you should have gofmt integrated in your editor.

**Pull request merged**